subprojects/configuration-cache/src/test/kotlin/org/gradle/configurationcache/serialization/codecs/JavaObjectSerializationCodecTest.kt

```diff
@@ -143,7 +143,7 @@ class JavaObjectSerializationCodecTest : AbstractUserTypeCodecTest() {
     @Ignore("wip")
     @Test
     fun `can handle Externalizable beans`() {
-        verifyRoundtripOf(::ExternalizableBean) {
+        verifyRoundtripOf({ ExternalizableBean() }) {
```

Comment:
😮  Looks like a compiler bug to me. [KT-42055](https://youtrack.jetbrains.com/issue/KT-42055) looks related.

Reply:
It definitely looks like this issue. I commented there